aicsimageio/readers/default_reader.py

```diff
@@ -2,86 +2,107 @@
+                        operating_shape = (image_length, ) + ((1, ) * len(sample.shape))
+                        # Create numpy array of empty arrays for delayed get data functions
+                        lazy_arrays = np.ndarray(operating_shape, dtype=object)
+                        for indicies, _ in np.ndenumerate(lazy_arrays):
+                            lazy_arrays[indicies] = da.from_delayed(
+                                delayed(self._get_data)(self._file, indicies[0]),
+                                shape=sample.shape,
+                                dtype=sample.dtype
+                            )
+
+                        # Block them into a single dask array
+                        self._dask_data = da.block(lazy_arrays.tolist())
+
+                    # Catch all other image types as unsupported
+                    # https://imageio.readthedocs.io/en/stable/userapi.html#imageio.core.format.Reader.get_length
+                    else:
+                        raise ValueError()
+
+            # Catch not a supported image
+            except ValueError:
+                raise exceptions.UnsupportedFileFormatError(self._file)
+
+        return self._dask_data
 
     @property
     def dims(self) -> str:
-        """
-        `imageio.imread` returns the first YX plane of an image, except in the case where the image is RGB / RGBA, in
-        which case it returns the first YXC plane of an image.
-
-        This dims property handles those cases by making assumptions about the dimension order based off the shape of
-        the data stored in the reader. In the case where more data than an YXC plane is returned, it uses the
-        `guess_dim_order` function. However, I can't tell when that should ever get run and is really just used as a
-        safety catch all.
-        """
+        # Set dims if not set
         if self._dims is None:
-            if len(self.data.shape) == 2:
+            if len(self.dask_data.shape) == 2:
                 self._dims = "YX"
-            elif len(self.data.shape) == 3:
+            elif len(self.dask_data.shape) == 3:
                 self._dims = "YXC"
+            elif len(self.dask_data.shape) == 4:
+                self._dims = "TYXC"
             else:
-                self._dims = self.guess_dim_order(self.data.shape)
+                self._dims = self.guess_dim_order(self.dask_data.shape)
 
         return self._dims
 
-    @dims.setter
-    def dims(self, dims: str):
-        # Check amount of provided dims against data shape
-        if len(dims) != len(self.data.shape):
-            raise exceptions.InvalidDimensionOrderingError(
-                f"Provided too many dimensions for the associated file. "
-                f"Received {len(dims)} dimensions [dims: {dims}] "
-                f"for image with {len(self.data.shape)} dimensions [shape: {self.data.shape}]."
-            )
-
-        # Set the dims
-        self._dims = dims
```

Comment:
Add this back but move it to the Reader base class

Reply:
Can't and shouldn't add it back to the Reader base class because certain readers like CZI and OME we shouldn't override the dims read from the metadata